oonib/report/handlers.py

```diff
@@ -192,27 +140,63 @@ def parseNewReportRequest(request):
 
     return parsed_request
 
-class ReportHandler(OONIBHandler):
+def closeReport(report_id):
+    report_dir = FilePath(config.main.report_dir)
+    archive_dir = FilePath(config.main.archive_dir)
+    report_path = report_dir.child(report_id)
+    report_metadata_path = report_dir.child(report_id +
+                                            "-metadata.json")
+    print "Checking if %s exists and %s exists" % (report_path,
+                                                   report_metadata_path)
+    if not report_path.exists() or \
+            not report_metadata_path.exists():
+        raise e.ReportNotFound
+
+    with report_metadata_path.open('r') as f:
+        report_details = json.load(f)
```

Comment:
This is a fair point, however if the metadata file is corrupt we actually never end up cleaning it.

I think that this sort of race condition is not very likely to occur given that in order for it to happen you must have created a report without ever writing data to it.

I am going to handle this condition by deleting the metadata file as well as the report file and logging this with level warning.